pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -91,7 +91,12 @@ func (util *VsphereDiskUtil) CreateVolume(v *vsphereVolumeProvisioner, selectedN
 		return nil, err
 	}
 	volSizeKiB := volSizeMiB * 1024
-	name := volumeutil.GenerateVolumeName(v.options.ClusterName, v.options.PVName, 255)
+	// reduce number of characters in vsphere volume name. The reason for setting length smaller than 255 is because typically
+	// volume name also becomes part of mount path - /var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/<name>
+	// and systemd has a limit of 256 chars in a unit name - https://github.com/systemd/systemd/pull/14294
+	// so if we subtract the kubelet path prefix from 256, we are left with 191 characters but because some characters
+	// will be escaped we are deliberating choosing a lower value of 180.
+	name := volumeutil.GenerateVolumeName(v.options.ClusterName, v.options.PVName, 180)
```

Comment:
IMO the name must be much shorter than 180. We add `[<datastore>] - <uuid>/` to it, where every `"[]-"` character is escaped as 4 characters by systemd.

For example, if we leave 20 characters for datastore names, such volume is mounted to 
```
/var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/[12345678901234567890] 31dc869f-aae9-4580-9d63-9461241123b0/<name>
```
That leaves 109 characters for `<name>` itself.  Would be `90` a better number? That allows users to use datastore names with 39 characters (or less, if they use dashes that are escaped).

Reply:
+1 on truncating the name a bit more (as we have the uuid) and leave more room to the datastore name

Reply:
+1 on truncating the name a bit more (as we have the uuid) and leave more room to the datastore name

Reply:
Is GenerateVolumeName only called on Create? Only updating the expected name here can break compatibility with other callers that expect 255 characters still.

Reply:
Yes it is only called on create and nowhere else. Updated the length to be 90 characters.

Reply:
Does changing the name here impact things like reconstruction/upgrades?

Reply:
nope. The volumeName is stored and always retrieved from PV and since we aren't changing name of existing PVs we should be fine.